### DISCUSSION TITLE
add/remove/sync device type updates to device

### DISCUSSION THREAD

**Discussion by Jackbennett** [netbox-community/netbox#6682] [Ideas]
Propose to add  to Device edit UI buttons to import/drop specific or all fields from the associated device type. Manually, not automatically.

1. Make your switch device type
2. Make 5 switches, rack them up.
3. Realise you could put all e.g 48 interfaces + 1 management on the device type.
4. Now you have to add those interfaces to each switch to match the device type so the 6th switch someone makes is consistent. 
5. propose: Edit device -> sync from device type to add what was define in the device Type.

I can see why you wouldn't want editing the device type to affect the device automatically. That'd get annoying and I wouldn't want to remove the interfaces from the type only to find all my devices lost them! But it would be really hand just to have a "match device type" button. Or for example if someone adds a power interface it can offer "add from device type"

Either a simple "import new from device type fields" button to save some work, or a full "here's the device and device type difference" add/remove picker ui.

Without resorting to the API the fastest way to do this is delete/recreate the device. Seems like a UX anti-pattern.

I looked for existing issues and didn't see this suggestion, but sorry if it has come up.

**Comment by candlerb** [comment]
What you need is a [custom script](https://netbox.readthedocs.io/en/stable/additional-features/custom-scripts/) to do it.

There's a CLI script to do it globally [here](https://gist.github.com/candlerb/9616db031301ec8fc737c620881d2fb8) - it was written before custom scripts were available, but it could be converted to one pretty easily.

There is also a report [here](https://gist.github.com/candlerb/d78886b93b36e570e1098eb1d346b76e) to identify devices with missing components.
